### PR TITLE
Change ttRPC unix socket path

### LINKED ISSUE #210: ttrpc | Change path of unix socket
It is suggested to change unix socket `unix:///opt/confidential-containers/attestation-agent/getresource.sock` and `"unix:///opt/confidential-containers/attestation-agent/keyprovider.sock"` to `/run/*` as they are temporary data at runtime.

cc @jialez0

### PR DESCRIPTION
**Pull request by jialez0**
Resolve: confidential-containers/guest-components#210

### CODE REVIEW

File: app/src/ttrpc.rs [hunk 1]
```diff
@@ -36,13 +36,18 @@ pub fn ttrpc_main() {
...
             .get_matches();
 
+    if !Path::new(DEFAULT_UNIX_SOCKET_DIR).exists() {
+        std::fs::create_dir_all(DEFAULT_UNIX_SOCKET_DIR).expect("Create unix socket dir failed");
+    }
+    let keyprovider_socket_file = format!("{UNIX_SOCKET_PREFIX}{DEFAULT_UNIX_SOCKET_DIR}/{KEYPROVIDER_SOCKET_FILE}");
     let keyprovider_socket = app_matches
         .value_of("KeyProvider ttRPC Unix socket addr")
-        .unwrap_or(DEFAULT_KEYPROVIDER_UNIX_ADDR);
+        .unwrap_or(&keyprovider_socket_file);
```
Comment (Xynnn007) [inline app/src/ttrpc.rs:45]: We might directly define two consts `keyprovider_socket_file` and `getresource_socket_file`, by which we can avoid concat the two strings in runtime

Comment (Xynnn007) [reply app/src/ttrpc.rs:45]: s.t. I prefer the old way

**Review by Xynnn007 — APPROVED** [review]
LGTM

**Review by jiangliu — APPROVED** [review]

File: app/src/ttrpc.rs [hunk 2]
```diff
@@ -7,10 +7,9 @@ use super::*;
 use ::ttrpc::Server;
 use std::path::Path;
 
-const DEFAULT_KEYPROVIDER_UNIX_ADDR: &str =
-    "unix:///opt/confidential-containers/attestation-agent/keyprovider.sock";
-const DEFAULT_GETRESOURCE_UNIX_ADDR: &str =
-    "unix:///opt/confidential-containers/attestation-agent/getresource.sock";
+const DEFAULT_UNIX_SOCKET_DIR: &str = "/run/confidential-containers/attestation-agent";
+const DEFAULT_KEYPROVIDER_SOCKET_ADDR: &str = "unix:///run/confidential-containers/attestation-agent/keyprovider.sock";
```
Comment (sameo) [inline app/src/ttrpc.rs:11]: Maybe we could use https://crates.io/crates/const_format and avoid duplicating strings?

**Review by sameo — APPROVED** [review]

**Pull request merged**